.github/workflows/run-checks.yml

```diff
@@ -0,0 +1,83 @@
+name: "Run Checks"
+
+on:
+  pull_request:
```

Comment:
Chore: could we also run the checks on pushes to the `main` branch in case of divergence during merges?